core/controllers/collection_viewer.py

```diff
@@ -0,0 +1,133 @@
+from core.domain import exp_services
+from core.domain import rights_manager
+from core.platform import models
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+import feconf
+import utils
+
+def require_collection_playable(handler):
+    """Decorator that checks if the user can play the given collection."""
+    def test_can_play(self, collection_id, **kwargs):
+        """Check if the current user can play the collection."""
+        if rights_manager.Actor(self.user_id).can_play(
+                rights_manager.ACTIVITY_TYPE_COLLECTION, collection_id):
+            return handler(self, collection_id, **kwargs)
+        else:
+            raise self.PageNotFoundException
+
+    return test_can_play
+
+
+class CollectionPage(base.BaseHandler):
+    """Page describing a single collection."""
+
+    PAGE_NAME_FOR_CSRF = 'collection'
+
+    @require_collection_playable
+    def get(self, collection_id):
+        """Handles GET requests."""
+        try:
+            collection = collection_services.get_collection_by_id(
+                collection_id)
+        except Exception as e:
+            raise self.PageNotFoundException(e)
+
+        version = collection.version
+
+        if not rights_manager.Actor(self.user_id).can_view(
+                rights_manager.ACTIVITY_TYPE_COLLECTION, collection_id):
+            raise self.PageNotFoundException
+
+        self.values.update({
+            'can_edit': (
+                bool(self.username) and
+                self.username not in config_domain.BANNED_USERNAMES.value and
+                rights_manager.Actor(self.user_id).can_edit(
+                    rights_manager.ACTIVITY_TYPE_COLLECTION, collection_id)
+            ),
+            'collection_id': collection_id,
+            'collection_title': collection.title,
+            'is_private': rights_manager.is_collection_private(collection_id),
+            'meta_name': collection.title,
+            'meta_description': utils.make_first_letter_uppercase(
+                collection.objective)
+        })
+
+        self.render_template('collection_player/collection_player.html')
+
+
+class CollectionDataHandler(base.BaseHandler):
+    """Provides the data for a single collection."""
+
+    def get(self, collection_id):
+        """Populates the data on the individual collection page."""
+        try:
+            collection = collection_services.get_collection_by_id(
+                collection_id)
+        except Exception as e:
+            raise self.PageNotFoundException(e)
+
+        exp_ids = [
+            collection_node.exploration_id
+            for collection_node in collection.nodes]
+
+        exp_summaries = (
+            exp_services.get_exploration_summaries_matching_ids(exp_ids))
+
+        exp_titles_dict = {}
+        for (ind, exp_id) in enumerate(exp_ids):
+            exp_summary = exp_summaries[ind]
+            exp_titles_dict[exp_id] = exp_summary.title if exp_summary else ''
```

Comment:
I suppose the question then is, which pages should possibly have out of date exploration meta data? If the summary can potentially be out of sync with the exp model, then the gallery might display wrong information. Why is it more important for a collection to display the correct information than the gallery? From my point of view, summaries should be fully reliable since there are places in which we completely rely on them. I like your idea of splitting up exploration model, but it seems a bit late for that. I think in the mean time, we can consider summaries to be an importantly up-to-date thin version of an exploration.